src/webots/maths/WbImage.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 1996-2018 Cyberbotics Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "WbImage.hpp"
+
+// #include <cmath>
+// #include <cstdio>
+#include <cstdlib>
+#include <cstring>
+
+#define CLAMP(v, low, high) ((v) < (low) ? (low) : ((v) > (high) ? (high) : (v)))
+
+static const float gaussBlur[3][3] = {
+  {1.0f / 16.0f, 2.0f / 16.0f, 1.0f / 16.0f},
+  {2.0f / 16.0f, 4.0f / 16.0f, 2.0f / 16.0f},
+  {1.0f / 16.0f, 2.0f / 16.0f, 1.0f / 16.0f},
+};
+
+WbImage *WbImage::downscale(int width, int height, int xBlurRadius, int yBlurRadius) {
+  const int dstSize = width * height * mChannels;
+  const int srcSize = mWidth * mHeight * mChannels;
+  unsigned char *pixels = (unsigned char *)malloc(dstSize);
+  memset(pixels, 0, dstSize);
```

Comment:
This is probably useless and CPU time consuming, isn't it?

Reply:
Yes, but it's not possible to avoid to create the new scaled-down image at some point, and as the following loop increment the elements, I need to initialize it. It's for sure anecdotical relatively to the following loop.